pkg/controller/installation/installation_controller.go

```diff
@@ -386,51 +315,157 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 	metrics.SetRHMIStatus(installation)
 
 	// Check if the version needs to be updated
-	if (isFirstInstallReconcile(installation) || isUpgradeReconcile(installation)) && allProductsReconciled {
+	if (isFirstInstallReconcile(installation) || isUpgradeReconcile(installation)) && allProductsReconciled(installation) {
+		installation.Status.Version = installation.Status.ToVersion
+		installation.Status.ToVersion = ""
+		metrics.SetRhmiVersions(string(installation.Status.Stage), installation.Status.Version, installation.Status.ToVersion, installation.CreationTimestamp.Unix())
+	}
+	metrics.SetRHMIStatus(installation)
+
+	if (isFirstInstallReconcile(installation) || isUpgradeReconcile(installation)) && allProductsReconciled(installation) {
```

Comment:
redundant? This seems to be the same condition as above with the same actions when true.

Reply:
this looks like a rebasing error. @CathalOConnorRH can we get the duplicate removed here?